conductor_api/src/conductor/base.rs

```diff
@@ -626,6 +626,59 @@ impl Conductor {
                     ))
                 })?;
 
+                // This is where we are checking the consistency between DNAs: for now we compare
+                // the hash provided in the TOML Conductor config file with the computed hash of
+                // the loaded dna.
+                {
+                    let dna_hash_from_conductor_config =
+                        HashString::from(dna_config.hash.clone()
+                                         // .unwrap_or("Missing DNA hash from Conductor config.".to_string()))
+                                         .unwrap_or("".to_string()))
+                                         .to_owned();
+                    let dna_hash_computed = &dna.address();
+
+                    // Let's warn the user that we encounter an empty DNA hash in the Conductor
+                    // config that is mmeant to be skipped by the consistency checking mechanisim
+                    context.log("warn/Conductor: Empty DNA hash encountered in the Conductor configuration.");
+
+                    match Conductor::load_dna(&dna_file) {
```

Comment:
`Conductor::load_dna` is the default `DnaLoader` that gets set if there is no other `DnaLoader` provided with the `Conductor` constructor. But the actual `DnaLoader` to use is stored in `Conductor::dna_loader`. We should definitely use the latter to not break the case where we do not use the default (file) loader - such as tests and the upcoming WASM conductor where we don't have a filesystem.

I wonder how this can work. I would expect tests to currently not really tests the code below.

This line should be:
```suggestion
                    match Arc::get_mut(&mut self.dna_loader).unwrap()(&dna_file) {
```

Reply:
Thank you for your precious input on the matter, I made the change accordingly :+1: